utils/check_repo.py

```diff
@@ -596,6 +597,33 @@ def check_all_objects_are_documented():
             + "\n - ".join(undocumented_objs)
         )
     check_docstrings_are_in_md()
+    check_model_type_doc_match()
+
+
+def check_model_type_doc_match():
+    """Check all doc pages have a corresponding model type."""
+    model_doc_folder = Path(PATH_TO_DOC) / "model_doc"
+    model_docs = [m.stem for m in model_doc_folder.glob("*.mdx")]
+
+    model_types = list(transformers.models.auto.configuration_auto.MODEL_NAMES_MAPPING.keys())
+
+    errors = []
+    for m in model_docs:
+        if m not in model_types and m != "auto":
```

Comment:
@sgugger I've added this `m != "auto"` clause as otherwise the script would complain that `auto` was not in the `MODEL_NAMES_MAPPING`. I don't think adding `auto` there makes sense so I've added this workaround here.

Will merge like this to prevent further conflicts, but open to discussion if you think another solution should have been implemented.

Reply:
Sounds good, thanks for fixing!